### ISSUE TITLE
Error When Launching Some Games From Sunshine

### ISSUE THREAD

**Issue by Silas888** [LizardByte/Sunshine#764]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your issue described in the documentation?

- [X] I have read the documentation

### Is your issue present in the nightly release?

- [X] This issue is present in the nightly release

### Describe the Bug

There is a when attempting to launch some games within Moonlight from Sunshine. The majority of the time I'll get the following error: "Connection Error Server Error." I'll take for example God of War. I have it configured in Sunshine the same way I have it in GeForce Experience. In the command section of Sunshine I have listed: "D:\Games\God of War\GoW.exe". If I move the same line to Detached Commands the desktop will show but the game won't be launched. Funny thing is that I also have Playnight configured and all games work when I launch them through Playnight . I'm thinking that this must then be related to how Sunshine launches the exe. Below is what I see if I look in the log file. The same log entry occurs for every game that fails to launch.

[2023:01:11:17:27:14]: Info: Executing: ["D:\Games\God of War\GoW.exe"] in ["D:\Games\God of War"]
[2023:01:11:17:27:14]: Error: Failed to launch process: 740
[2023:01:11:17:27:14]: Warning: Couldn't run ["D:\Games\God of War\GoW.exe"]: System: Invalid argument

### Expected Behavior

The expected behaviors for the game to launch

### Host Operating System

Windows

### Operating System Version

Windows 10 version 22H2 

### Architecture

64 bit

### Sunshine commit or version

v0.17.0

### Package

Windows - installer

### GPU Type

Nvidia

### GPU Model

GTX 1080 Ti

### GPU Driver/Mesa Version

Driver Version: 528.02

**Comment by jjohnson338** [comment]
I've got a similar issue with and I've found a temporary workaround. I've created a `.bat` script and placed it on the desktop:
```
START S:\SteamLibrary\steamapps\common\"ELDEN RING"\Game\eldenring.exe
```

I then add the path to the bat script as a Detached Command (with no regular command). This ends up launching the game and starting the stream.

**Comment by Silas888** [comment]
Thanks for the suggestion. I actually setup Playnite and through that application all the games work. I suppose I can just use that for everything, but I kinda wanted to work through Moonlight too because that's how it worked with Game Stream

**Comment by Nonary** [comment]
> I've got a similar issue with and I've found a temporary workaround. I've created a `.bat` script and placed it on the desktop:
> 
> ```
> START S:\SteamLibrary\steamapps\common\"ELDEN RING"\Game\eldenring.exe
> ```
> 
> I then add the path to the bat script as a Detached Command (with no regular command). This ends up launching the game and starting the stream.

That's because currently de-attached commands are running with system level privileges. But that will be fixed in upcoming versions of Sunshine when #998 is resolved.

As for this issue specifically, if the game require elevation you will not be able to launch it with Sunshine.

For security reasons, Sunshine will execute commands without elevation.

The main reason for this is because the configuration file does not require administration rights to modify. So we can't just add a simple elevated checkbox to the configuration, since it doesn't require admin rights to save it...

Long story short a hacker with knowledge of sunshine could take control of a machine with just normal user rights.

Fixing this problem is complicated, but really needs to be a feature request if people want it implemented.
https://github.com/LizardByte/Sunshine/discussions/categories/feature-requests

In the meantime, you can use https://www.winhelponline.com/blog/run-programs-elevated-without-getting-the-uac-prompt/ as a workaround.
